### PR TITLE
Android UI for HTTPS by Default

### PR DESCRIPTION
**Pull request by arthuredelstein**
- global pref UI
- per-site setting UI

Resolves https://github.com/brave/brave-browser/issues/28295

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java [hunk 1]
```diff
@@ -741,12 +760,17 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
 
     private void setUpSwitchLayouts() {
-
         LinearLayout mUpgradeHttpsLayout = mSecondaryLayout.findViewById(R.id.brave_shields_upgrade_https_id);
-        TextView mUpgradeHttpsText = mUpgradeHttpsLayout.findViewById(R.id.brave_shields_switch_text);
-        mBraveShieldsHTTPSEverywhereSwitch = mUpgradeHttpsLayout.findViewById(R.id.brave_shields_switch);
-        mUpgradeHttpsText.setText(R.string.brave_shields_https_everywhere_switch);
-        setupHTTPSEverywhereSwitchClick(mBraveShieldsHTTPSEverywhereSwitch);
+        if (!ChromeFeatureList.isEnabled(BraveFeatureList.HTTPS_BY_DEFAULT)) {
+            TextView mUpgradeHttpsText =
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java:765]: nit : local variables can be `upgradeHttpsText`

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -572,6 +572,24 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
       <message name="IDS_BLOCK_TRACKERS_ADS_SUMMARY" desc="Text summary for Block trackers and ads.">
         Prevents trackers &amp; ads from loading and showing in webpages
       </message>
+      <message name="IDS_HTTPS_UPGRADE_TEXT" desc="Text for shields option.">
```
Comment (deeppandya) [inline browser/ui/android/strings/android_brave_strings.grd:575]: we should use descriptive `desc` as it's useful for translation.

**Review by deeppandya — APPROVED** [review]
LGTM. a couple of minor comments

File: android/java/org/chromium/chrome/browser/preferences/website/BraveShieldsContentSettings.java [hunk 3]
```diff
@@ -91,6 +92,9 @@ public static void setShieldsValue(Profile profile, String host, String resource
         if (resourceIndentifier.equals(RESOURCE_IDENTIFIER_FINGERPRINTING)) {
             BraveShieldsContentSettingsJni.get().setFingerprintingControlType(
                     settingOption, host, profile);
+        } else if (resourceIndentifier.equals(RESOURCE_IDENTIFIER_HTTPS_UPGRADE)) {
+            BraveShieldsContentSettingsJni.get().setHttpsUpgradeControlType(
+                    settingOption, host, profile);
```
Comment (wchen342) [inline android/java/org/chromium/chrome/browser/preferences/website/BraveShieldsContentSettings.java:97]: Is https://github.com/brave/brave-core/blob/a8fa5589278dbf11033260aea82c72e12bed26cc/android/java/org/chromium/chrome/browser/preferences/website/BraveShieldsContentSettings.java#L84 still needed if we have this HTTPS upgrade setting now?

**Review by wchen342 — APPROVED** [review]
++, can you please add a test plan for the new UI?

File: browser/ui/android/strings/android_brave_strings.grd (hunk 2 above)
Comment (arthuredelstein) [reply browser/ui/android/strings/android_brave_strings.grd]: fixed

**Comment by arthuredelstein** [comment]
**QA Testing ideas**

A flag (https-by-default) is needed to turn on this feature. If the flag is disabled, then the normal https-everywhere behavior should obtain.

There are three places where the user can control HTTPS Upgrades:

1. Global preference (3 states: disabled, standard, strict)
2. The per-site Shields toggle (2 states: up, down)
3. The per-site HTTPS upgrades control (3 states: disabled, standard, strict)

As with other shields-related controls, per-site should override global. In addition, the default setting is standard, but if the user changes a setting it should persist.

Expected behaviors are:

|               | Disabled      | Standard      | Strict |
| ------------- | ------------- | ------------- | ------ |
| http://upgradable.arthuredelstein.net | No upgrade | Upgrade | Upgrade |
| http://insecure.arthuredelstein.net   | No upgrade  | No upgrade | Insecure site error page |

File: android/java/org/chromium/chrome/browser/preferences/website/BraveShieldsContentSettings.java (hunk 3 above)
Comment (arthuredelstein) [reply android/java/org/chromium/chrome/browser/preferences/website/BraveShieldsContentSettings.java:97]: We will be removing HTTPS Everywhere support in the future, once this is fully deployed.

**Review by iefremov — APPROVED** [review]
chromium_src lgtm

**Pull request merged**